libp2p/network/url_conversion.go

```diff
@@ -24,7 +26,11 @@ const (
 	errResolvePeerIPMsg = "resolving peer IP for hostname"
 )
 
-// Libp2pMultiaddrFromServiceURL transforms a URL into its libp2p multiaddr equivalent.
+// Libp2pMultiaddrFromServiceURL transforms a URL into its libp2p multiaddr equivalent. The URL must contain a port number.
+// The URL may contain a hostname or IP address. If a hostname is provided, it will be resolved to an IP address.
+// The URL may not contain a scheme. The URL will be prefixed with "scheme://".
+// The URL may contain an IPv6 address, but it must be enclosed in square brackets.
+// The URL may contain an IPv4 address, but it must not be enclosed in square brackets.
```

Comment:
I think it would be helpful to also reference the RFC here with a line like so:
```suggestion
// The URL may contain an IPv6 address, but it must be enclosed in square brackets.
// The URL may contain an IPv4 address, but it must not be enclosed in square brackets.
(see: https://www.rfc-editor.org/rfc/rfc3986#section-3.2.2)
```